subsys/bluetooth/mesh/net.c

```diff
@@ -1329,7 +1324,7 @@ void bt_mesh_net_recv(struct net_buf_simple *data, s8_t rssi,
 	 */
 	if (bt_mesh_trans_recv(&buf, &rx) == -EAGAIN) {
 		BT_WARN("Removing rejected message from Network Message Cache");
-		msg_cache[rx.msg_cache_idx] = 0ULL;
+		msg_cache[rx.msg_cache_idx].src = BT_MESH_ADDR_UNASSIGNED;
 		/* Rewind the next index now that we're not using this entry */
 		msg_cache_next = rx.msg_cache_idx;
```

Comment:
@jhedberg I have a feeling this `msg_cache_idx` member could now be removed, and this line changed to `msg_cache_next = (msg_cache_next - 1) % ARRAY_SIZE(msg_cache)`, as the packet now only exists in a single cache entry. I don't know if I might be missing some preemption issue in case this happens on the system work queue, and the variable doesn't take any space in the rx structure, so I left it in. Opinions?